### PR TITLE
Feature/delete target

### PR DESCRIPTION
**Pull request by peter7z**
Targets can now be selected in order to be deleted.
The API response for the delete target endpoint has status 200 and an empty body, so a small fix has been made in order to avoid a promise rejection in "apiService.js".
screen shot 2018-10-09 at 3 47 40 pm

### CODE REVIEW

File: src/actions/targetActions.js [hunk 1]
```diff
@@ -10,16 +10,22 @@ const getTargetsSuccess = ({ targets }) => ({
...
     .then(targets => dispatch(getTargetsSuccess(targets)))
     .catch(() => toast.error('Could not retrieve targets'));
 };
 
-export const createTarget = target => () => {
+export const createTarget = target => (dispatch) => {
   history.push(routes.index);
   targetApi.createTarget(target)
     .then(() => {
+      dispatch(getTargets());
```
Comment (felipegarcia92) [inline src/actions/targetActions.js:28]: Normally, apart from this, you would append the new target to your targets list, so the response in the front-end is immediate.

So, you can have and dispatch a `createTargetSuccess` action  with the newly created target as payload, and use that to append it to the targets list you have in the reducer. I know the response times are quite quick here, but what about a slow network?

File: src/api/apiService.js [hunk 2]
```diff
@@ -43,20 +45,22 @@ const handleErrors = response =>
...
 
 class Api {
-  performRequest(uri, apiUrl, requestData = {}) {
+  performRequest(uri, apiUrl, requestData = {}, del = false) {
     const url = `${apiUrl}${uri}`;
     return new Promise((resolve, reject) => {
       fetch(url, requestData)
         .then(handleErrors)
-        .then(getResponseBody)
+        .then(!del ? getResponseBody : empty)
```
Comment (felipegarcia92) [inline src/api/apiService.js]: What's del? Delete? Can you name it with the full description?

File: src/containers/HomePage.js [hunk 3]
```diff
@@ -36,18 +38,37 @@ class HomePage extends React.Component {
...
     if (this.props.match.isExact) history.push(routes.newTarget);
   }
 
   render() {
     const { location, subRoutes, targets } = this.props;
-    const { target, coords } = this.state;
+    const { target, coords, selectedTarget } = this.state;
     return (
       <div className="home-page">
+        {this.state.selectedTarget != undefined &&
```
Comment (felipegarcia92) [inline src/containers/HomePage.js]: You are already taking `selectedTarget` from `this.state` destructuring above.

Also, `selectedTarget != undefined` would be the same to just `selectedTarget &&`

**Review by felipegarcia92 — CHANGES_REQUESTED** [review]
Good handling of redux and components state 👍

File: src/actions/targetActions.js (hunk 1 above)
Comment (peter7z) [reply src/actions/targetActions.js:28]: I thought about that, but what if then the user wants to delete the target? A target ID is needed, which I don't have.

File: src/containers/HomePage.js (hunk 3 above)
Comment (peter7z) [reply src/containers/HomePage.js]: selectedTarget is an array index. When the selection is the first item in the array it's 0 and evaluates to false as a boolean.

File: src/actions/targetActions.js (hunk 1 above)
Comment (felipegarcia92) [reply src/actions/targetActions.js:28]: Good point 👌

**Review by felipegarcia92 — APPROVED** [review]

File: src/containers/HomePage.js (hunk 3 above)
Comment (felipegarcia92) [reply src/containers/HomePage.js]: Makes sense

File: src/api/apiService.js [hunk 4]
```diff
@@ -43,20 +45,22 @@ const handleErrors = response =>
   });
 
 const getResponseBody = (response) => {
-  const bodyIsEmpty = response.status === 204;
+  console.log(response.body);
```
Comment (xlasserre) [inline src/api/apiService.js]: Remove console.log

File: src/components/common/Map.js [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
 import { withScriptjs, withGoogleMap, GoogleMap, Marker, Circle } from 'react-google-maps';
-import { object, bool, func, array } from 'prop-types';
+import { object, bool, func, array, number } from 'prop-types';
 
 import marker from '../../img/marker.svg';
-import { macCheeseColor } from '../../constants/styleConstants';
+import { macCheeseColor, white } from '../../constants/styleConstants';
 
-const Map = ({ targets, handleClick, isMarkerShown, coords: { latitude, longitude }, target }) => (
+const Map = ({ selectedTarget, selectTarget, targets, handleClick, isMarkerShown, coords: { latitude, longitude }, target }) => (
```
Comment (xlasserre) [inline src/components/common/Map.js]: Could you shorten this line?

**Review by xlasserre — CHANGES_REQUESTED** [review]
Good job! I only added two code styling comments. Btw, did you create an issue in the backend project for the delete response error you found?

**Review by xlasserre — APPROVED** [review]

**Pull request merged**